### PR TITLE
 Allow application to provide predefined log configuration

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Comment by findepi** [comment]
cc @electrum @dain @kokosing

**Comment by findepi** [comment]
I am yet to verify this actually would work with Presto as expected. I want to know your opinion about general idea first.

File: log-manager/src/main/java/io/airlift/log/Logging.java [hunk 1]
```diff
@@ -201,6 +217,8 @@ public void configure(LoggingConfiguration config)
             disableConsole();
         }
 
+        setBuiltInLevels();
```
Comment (kokosing) [inline log-manager/src/main/java/io/airlift/log/Logging.java]: shouldn't built in levels be loaded before user log levels?

File: log-manager/src/main/java/io/airlift/log/Logging.java [hunk 2]
```diff
@@ -128,12 +129,27 @@ public void setRootLevel(Level newLevel)
         setLevel(ROOT_LOGGER_NAME, newLevel);
     }
 
+    private void setBuiltInLevels()
```
Comment (kokosing) [inline log-manager/src/main/java/io/airlift/log/Logging.java]: move it under the usage

File: log-manager/src/main/java/io/airlift/log/Logging.java (hunk 1 above)
Comment (findepi) [reply log-manager/src/main/java/io/airlift/log/Logging.java]: They should. Aren't they?

File: log-manager/src/main/java/io/airlift/log/Logging.java (hunk 2 above)
Comment (findepi) [reply log-manager/src/main/java/io/airlift/log/Logging.java]: Please see the whole file, not just the diff. This was IMO the most logical place.

**Comment by electrum** [comment]
For the Hadoop log levels in Presto, @martint and I discussed adding an API for connectors to set the default levels.

Loading from a resource won’t work for Presto since we want to set defaults inside a plugin, but logging is configured in the main server. We wouldn’t want to ship plugin-specific defaults in the main server.

**Comment by electrum** [comment]
Do we still want the first commit?

**Comment by findepi** [comment]
I forgot about the first commit... Do you want it?

**Comment by electrum** [comment]
Yes, let's fix it. It seems unlikely to matter in practice, but we shouldn't ever depend on the platform encoding.

**Comment by findepi** [comment]
@electrum https://github.com/airlift/airlift/pull/686

**Pull request closed without merging**